File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, h: List[int]) -> int:
        n = len(h)
        p,s = [h[0]],[h[n-1]]

        for i in range(1,n):
            if p[i-1]>h[i]:
                p.append(p[i-1])
            else:
                p.append(h[i])
            
            if s[i-1]>h[n-i-1]:
                s.append(s[i-1])
            else:
                s.append(h[n-i-1])
        s.reverse()
        ans = 0
        for i in range(n):
            ans += min(s[i],p[i])-h[i]
        return ans

File: Data Structures & Algorithms/climbing-stairs/submission-1.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n<=2) return n;
        int a = 1;
        int b = 2;
        while(n>2){
            int t = a+b;
            a = b;
            b = t;
            n--;
        }

        return b;
    }
};
